File: Exercise3/app/src/main/java/com/example/exercise3/MainActivity.java
package com.example.exercise3;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button go_btn = findViewById(R.id.button);
        go_btn.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, SecondActivity.class);
            startActivity(intent);
        });
    }
}

File: basketballScore/app/src/main/java/com/example/basketballscore/MainActivity.kt
package com.example.basketballscore

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.example.basketballscore.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private var myViewModel: MyViewModel? = null
    private var binding: ActivityMainBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        restoreScores()
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        myViewModel = ViewModelProvider(this)[MyViewModel::class.java]
        binding?.data = myViewModel
        binding?.lifecycleOwner = this
    }

    override fun onPause() {
        super.onPause()
        saveScores()
    }

    override fun onResume() {
        super.onResume()
        restoreScores()
    }

    private fun saveScores() {
        val sharedPref = getSharedPreferences("ScorePref", Context.MODE_PRIVATE)
        with(sharedPref.edit()) {
            putInt("TeamAScore", myViewModel?.getScoreTeamA()?.value ?: 0)
            putInt("TeamBScore", myViewModel?.getScoreTeamB()?.value ?: 0)
            apply()
        }
    }

    private fun restoreScores() {
        val sharedPref = getSharedPreferences("ScorePref", Context.MODE_PRIVATE)
        myViewModel?.getScoreTeamA()?.value = sharedPref.getInt("TeamAScore", 0)
        myViewModel?.getScoreTeamB()?.value = sharedPref.getInt("TeamBScore", 0)
    }
}